pip/req/req_set.py

```diff
@@ -204,6 +204,32 @@ def locate_files(self):
                     (req_to_install, req_to_install.source_dir)
                 )
 
+    @staticmethod
+    def _available_extras(provided, requested):
+        """
+        Given a tuple of installable extras, and a list of extras that the user
```

Comment:
It might be a tad clearer if you mentioned the argument names `provided` and `requested` here. It took me a few seconds to map the prose to the arguments.